zerver/lib/actions.py

```diff
@@ -363,6 +385,29 @@ def notify_new_user(user_profile: UserProfile) -> None:
         message = _("{user} just signed up for Zulip. (total: {user_count})").format(
             user=f"@_**{user_profile.full_name}|{user_profile.id}**", user_count=user_count
         )
+        plan = get_current_plan_by_realm(user_profile.realm)
+        if plan is not None and not plan.automanage_licenses:
+            licenses_remaining = plan.licenses() - get_latest_seat_count(user_profile.realm)
+            if licenses_remaining <= 3:
+                message += "\n"
+                if licenses_remaining <= 0:
+                    message += _(
+                        "Your organization has no spare Zulip licenses remaining and can no longer add new users."
+                    )
+                else:
+                    message += _(
+                        "Your organization has only {licenses_remaining} spare Zulip licenses remaining. This means your organization "
+                        "can add only {licenses_remaining} more users in the current billing period."
+                    ).format(licenses_remaining=licenses_remaining)
+                message += " "
+                message += _(
+                    "Please [increase the number of licenses]({billing_page_link}) or "
+                    "[deactivate inactive users]({deactivate_user_help_page_link}) to allow more members to join."
+                ).format(
+                    billing_page_link="/billing/#settings",
+                    deactivate_user_help_page_link="/help/deactivate-or-reactivate-a-user",
+                )
```

Comment:
I don't think we want to have this large/complex of a block of billing-specific logic in the `actions.py` code; we should move this to a function in `corporate/`.

Also `send_user_unable_to_signup_message_to_signup_notification_stream` probably belongs in `corporate/` somewhere as well.